### PR TITLE
Feat: plot summary with exception handling

### LINKED ISSUE #18: About function plot_summary
Function specific:
- [ ] Writing code for each function
- [ ] Handle incorrect input and errors with useful error messages
- [ ] Writing unit test function for each function (test_function_name.py in tests/ folder, 3-5 edge cases, with documentation, make sure that the tests cover all branches)
- [ ] use pytest to check test and code coverage

Feel free to add/edit/comment anything, or document any ideas/notes/test cases that do not have time to implement in this milestone.

### PR DESCRIPTION
**Pull request by ytz**
`TODO`

- [x] Use `get_data()` instead of `pd.read_csv(...)`

### CODE REVIEW

**Review by thomassiu — APPROVED** [review]

File: src/covizpy/plot_summary.py [hunk 1]
```diff
@@ -22,9 +27,68 @@ def plot_summary(
...
-        Start date for plot summary, by default '2022-01-01'
+        Start date of the data range with format '%Y-%m-%d'. By default 'None' is used to represent 7 days prior to today's date
     date_to : str, optional
-        End date for plot summary, by default '2022-01-13'
+        End date of data range with format '%Y-%m-%d'. By default 'None' is used to represent today's date
     top_n : int, optional
         Specify number of qualitative values to show, by default 5
     """
+    # Load data
+    df = pd.read_csv("https://covid.ourworldindata.org/data/owid-covid-data.csv")
```
Comment (rrrohit1) [inline src/covizpy/plot_summary.py]: Instead of checking below whether the df is present or not, you can check here if the link is broken or not.

`try:
        df = pd.read_csv('https://covid.ourworldindata.org/data/owid-covid-data.csv')
    except:
        return "The link to the data is broken"`

**Review by rrrohit1 — APPROVED** [review]

File: src/covizpy/plot_summary.py (hunk 1 above)
Comment (thomassiu) [reply src/covizpy/plot_summary.py]: I'll also add this exception check to my version in get_data.

File: src/covizpy/plot_summary.py [hunk 2]
```diff
@@ -22,9 +30,65 @@ def plot_summary(
     fun : str, optional
         Aggregation function for val, by default 'sum'
     date_from : str, optional
-        Start date for plot summary, by default '2022-01-01'
+        Start date of the data range with format '%Y-%m-%d'. By default 'None' is used to represent 7 days prior to today's date
```
Comment (thomassiu) [inline src/covizpy/plot_summary.py]:
```suggestion
        Start date of the data range with format 'YYYY-MM-DD'. By default 'None' is used to represent 7 days prior to today's date
```

Align using above for better understanding.

File: src/covizpy/plot_summary.py [hunk 3]
```diff
@@ -22,9 +30,65 @@ def plot_summary(
...
     date_to : str, optional
-        End date for plot summary, by default '2022-01-13'
+        End date of data range with format '%Y-%m-%d'. By default 'None' is used to represent today's date
```
Comment (thomassiu) [inline src/covizpy/plot_summary.py]:
```suggestion
        End date of data range with format 'YYYY-MM-DD'. By default 'None' is used to represent today's date
```
Align to use above for better understanding.

**Review by thomassiu — APPROVED** [review]
Reviewed with inline suggestions. I also updated mine in PR #26
